src/sentry/static/sentry/app/utils/discover/traceHoverCard.tsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import {Location, LocationDescriptor} from 'history';
+import styled from '@emotion/styled';
+
+import {t} from 'app/locale';
+import withApi from 'app/utils/withApi';
+import {Client} from 'app/api';
+import Hovercard from 'app/components/hovercard';
+import Version from 'app/components/version';
+import space from 'app/styles/space';
+import Clipboard from 'app/components/clipboard';
+import {IconCopy} from 'app/icons';
+import {TableData} from 'app/views/eventsV2/table/types';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import LoadingError from 'app/components/loadingError';
+
+import EventView from './eventView';
+import DiscoverQuery from './discoverQuery';
+
+type ChildrenProps = {to: LocationDescriptor};
+
+type Props = {
+  api: Client;
+
+  orgId: string;
+  traceId: string;
+
+  location: Location;
+
+  children: (props: ChildrenProps) => React.ReactNode;
+
+  // hover card props
+  containerClassName: string;
+};
+
+class TraceHoverCard extends React.Component<Props> {
+  renderHeader() {
+    const {traceId} = this.props;
+
+    return (
+      <HeaderWrapper>
+        <span>{t('Trace')}</span>
+        <TraceWrapper>
+          <StyledTrace version={traceId} truncate anchor={false} />
+          <Clipboard value={traceId}>
+            <ClipboardIconWrapper>
+              <IconCopy size="xs" />
+            </ClipboardIconWrapper>
+          </Clipboard>
+        </TraceWrapper>
+      </HeaderWrapper>
+    );
+  }
+
+  renderBody({
+    tableData,
+    isLoading,
+    error,
+  }: {
+    tableData: TableData | null;
+    isLoading: boolean;
+    error: null | string;
+  }) {
+    if (isLoading) {
+      return (
+        <LoadingWrapper>
+          <LoadingIndicator mini />
+        </LoadingWrapper>
+      );
+    }
+
+    if (error) {
+      return <LoadingError />;
+    }
+
+    if (!tableData) {
+      return null;
+    }
+
+    const numOfTransactions = tableData?.data.length ?? 0;
```

Comment:
If you update the results to only fetch the count this would be `tableData.data[0].length`